test/functional/test_framework/wallet.py

```diff
@@ -188,6 +188,38 @@ def send_to(self, *, from_node, scriptPubKey, amount, fee=1000):
         txid = self.sendrawtransaction(from_node=from_node, tx_hex=tx.serialize().hex())
         return txid, 1
 
+    def chain_tx(self, node, parent_txids, num_outputs=1, fee_per_output=1000):
+        """
+        Build and send a transaction that spends the given inputs and creates a
```

Comment:
```suggestion
        Build and send a transaction that spends one output of the given txids and creates a
```

I think it only spends the first found (random) output

Reply:
Maybe it would be better to get rid of this and put the burden of coin selection on the caller? This would also allow the caller to decide whether the inputs are marked as spent or not if the call fails. (Other methods here also mark them as spent by default, even on failure)

Reply:
Yes, I agree. Reworked the helper to accept and return UTXOs rather than txids now (i.e. caller has to do coin selection), which makes the test even shorter.